frontend/packages/git-service/src/services/hooks.ts

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecretKind } from '@console/internal/module/k8s';
+import { GitSource, SecretType as GitSecretType, GitProvider } from '../types';
+import { detectDeployStrategies, DetectedStrategy } from '../utils/deploy-strategy-detector';
+import { detectGitProvider } from '../utils/git-provider-detector';
+import { getGitService } from './git-service';
+
+type DetectedServiceData = {
+  loaded: boolean;
+  loadError?: any;
+  gitProvider: GitProvider;
+  strategies: DetectedStrategy[];
+};
+
+const gitServiceFromSource = (
```

Comment:
Shouldn't this just replace `getGitService`?

Reply:
+1, updated `getGitService` instead of wrapping over it.